src/main/java/orionmipt/Context.java

```diff
@@ -0,0 +1,309 @@
+package orionmipt;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+
+class Context implements Cloneable {
```

Comment:
Do not use `Cloneable` and do not mess with `clone` method overriding. This is a very rarely used feature which is dangerous and not recommended for use (see Effective Java Chapter 3 Item 13)